ophyd/device.py

```diff
@@ -303,7 +443,11 @@ def maybe_add_prefix(self, instance, kw, suffix):
         return suffix.format(**format_dict)
 
 
-class DynamicDeviceComponent(Component):
+component = Component.create
+formatted_component = FormattedComponent.create
```

Comment:
Are these the best names for these? It's ok for them to be slightly more verbose, I'm a bit worried about confusion resulting from  what amounts to just a case shift.

`typed_component`, maybe?

Reply:
I like the simplicity of the naming here, but I understand your hesitation. I'm open to input if anyone feels strongly on names.